message_parser/src/main/java/cumulus_message_adapter/message_parser/MessageAdapter.java

```diff
@@ -0,0 +1,127 @@
+package cumulus_message_adapter.message_parser;
+
+import com.amazonaws.services.lambda.runtime.Context; 
+
+import java.io.IOException;
+import java.io.OutputStreamWriter;
+import java.util.Scanner;
+import java.util.Map;
+import java.util.HashMap;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+
+public class MessageAdapter implements IMessageAdapter
+{
+    /**
+     * Call to message adapter zip to execute a message adapter function. Pass args through the process input
+     * and read return result from process output.
+     * @param messageAdapterFunction - 'loadRemoteEvent', 'loadNestedEvent', or 'createNextEvent'
+     * @param inputJson - argument to message adapter function. Json that contains all of the params.
+     * @return the return from the message adapter function
+     */
+    public String CallMessageAdapterFunction(String messageAdapterFunction, String inputJson)
+        throws MessageAdapterException
+    {
+        String messageAdapterOutput = "";
+
+        try
+        {
+            ProcessBuilder processBuilder = new ProcessBuilder("python", "cumulus-message-adapter.zip", messageAdapterFunction);
+
+            Process process = processBuilder.start();
+
+            OutputStreamWriter writer = new OutputStreamWriter(process.getOutputStream());
+            writer.write(inputJson);
+            writer.close();
+
+            // Log the entire error
+            // TO DO: Update logging
+            Scanner scanner = new Scanner(process.getErrorStream());
+            Boolean hasError = false;
+            while(scanner.hasNextLine()) 
+            {
+                hasError = true;
```

Comment:
This is using stderr to determine if there was an error.  It should be using the exit code instead.

Reply:
Changed it to check the exit code.